### ISSUE TITLE
Latest Gamescope breaks NVIDIA

### ISSUE THREAD

**Issue by kodatarule** [ValveSoftware/gamescope#987]
The latest version of gamescope has broken nvidia completely, all you see is a black window.

Operating System: EndeavourOS 
KDE Plasma Version: 5.27.8
KDE Frameworks Version: 5.110.0
Qt Version: 5.15.11
Kernel Version: 6.5.7-273-tkg-bmq (64-bit)
Graphics Platform: Wayland
Processors: 16 × AMD Ryzen 7 5800X3D 8-Core Processor
Memory: 31,3 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 3090/PCIe/SSE2

gamescope 3.12.7-1

**Comment by HunterCZ122** [comment]
Unfortunately, Gamescope is affected by an Nvidia driver bug that Josh needs to work around (#948). My recommendation is to use an older version of Gamescope until it gets fixed.
You can revert it on your Arch-based system by grabbing an older package from Arch Linux Archive:
`sudo pacman -U https://archive.archlinux.org/packages/g/gamescope/gamescope-3.12.5-1-x86_64.pkg.tar.zst`

**Comment by jarrard** [comment]
Yeah I noticed this issue with gamescope-plus package.

**Comment by Sweeper5** [comment]
Well you are in luck, since May this year, no package of Gamescope have worked for me under Nvidia.
Now to verify it wasn't Gamescope itself being faulty, I have tried the same versions on my laptop that is an all AMD setup with a Vega 3 GPU.
Suffice to say, it runs flawlessly and I can really see why people love Gamescope on there.

**Comment by HunterCZ122** [comment]
> Well you are in luck, since May this year, no package of Gamescope have worked for me under Nvidia.
> Now to verify it wasn't Gamescope itself being faulty, I have tried the same versions on my laptop that is an all AMD setup with a Vega 3 GPU.
> Suffice to say, it runs flawlessly and I can really see why people love Gamescope on there.

I also ran gamescope through Intel iGPU, which worked fine as a workaround until 545.
Hopefully NVK will give us some hope in the future if Nvidia doesn't fix this.

**Comment by ageisen2000** [comment]
I'm also getting this error on a 3090 on endeavouros. Let me know if I can send any logs or anything to help

**Comment by unhappy-ending** [comment]
Try the 535.43.20 vulkan beta branch, gamescope is working on that branch for me.

**Comment by jarrard** [comment]
just use gamescope 5.12, easier solution

**Comment by jeisom** [comment]
> Try the 535.43.20 vulkan beta branch, gamescope is working on that branch for me.

Still broke here with Vulkan beta x.x.20. Immediate segfault.  5.12.5 works though.

**Comment by unhappy-ending** [comment]
> > Try the 535.43.20 vulkan beta branch, gamescope is working on that branch for me.
> 
> Still broke here with Vulkan beta x.x.20. Immediate segfault. 5.12.5 works though.

Yeah, it beta drivers fixed some stuff on 5.13 but some games would outright refuse to launch. 5.12.5 was working yesterday, but now it's not working, no idea what changed so it's a little frustrating

**Comment by unhappy-ending** [comment]
Another update, 535 had intense horizontal screen tearing while playing games. I can't recommend it. 545 has major flickering in Chromium based web views which means Steam's overlay, but there isn't the screen tearing issue and Steam can use software rendering to "resolve" the disappearing graphics and flickering issue. Gamescope 5.12.5 is still working fine with 545, so it seems the best thing to do is stick to it until Nvidia resolves all the issues at hand.

**Comment by jarrard** [comment]
Some of the issues may just be wayland related if your using that (most are these days).  Still some rough edges for NVIDIA under Wayland which you need to work around.

**Comment by HunterCZ122** [comment]
If you guys get crash when launching any Vulkan game, you need to disable Gamescope WSI:
`ENABLE_GAMESCOPE_WSI=0 gamescope -- %command%`

**Comment by jarrard** [comment]
Still blackscreens games on nvidia and wayland

**Comment by thechitowncubs** [comment]
Total black or transparent window with nvidia 545

**Comment by AlexFolland** [comment]
I'm getting permanent freezes in the first minute with gamescope on Nvidia with Nvidia 545.29.06.  It's not a black screen, but it's still unplayable.

**Comment by thechitowncubs** [comment]
I pulled updated git and it works but permanent freeze issue here too

**Comment by peacey** [comment]
I am having a similar issue with gamescope freezing or having a blank screen when running a game. Using latest gamescope git and latest Nvidia 550 beta driver. 

The issue manifests slightly differently if running gamescope from within another compositor or standalone. 

1. If running gamescope from another compositor (like kwin), the game starts up and works for a few seconds to a couple minutes then freezes completely (only video freezes, audio remains playing in background so game is still running). The game never unfreezes, can't be exited sanely, and can only be force closed by killing the gamescope session.

2. If running gamescope in its own standalone session, the game starts up in a black window and never updates. The audio can be heard so game is running, but the window remains black forever. Again, only force killing gamescope can close the game. 

All these tests were done with only the minimal parameters "-W 1920 -H 1080 -f". Issue happens whether HDR is enabled or not.

_Sidenote: HDR gaming with Nvidia for the first time with the 550 driver was super cool for a hot minute before the game froze!_

**Comment by profpyrus** [comment]
As far as I can tell, this issue still persists, specifically trying to fix Fragpunk running in hyprland.
The game starts just fine, and when I try to join a match or go into the range, it works briefly before crashing. Any news on this?
